File: server/src/seed.ts
import 'dotenv/config'
import mongoose from 'mongoose'
import Problem from './models/Problem'

const MONGO_URL = process.env.MONGO_URL || 'mongodb://localhost:27017/oj'

const problems = [
	{
		title: 'A + B',
		description: 'Read two integers and output their sum.',
		inputFormat: 'Two integers a and b',
		outputFormat: 'Single integer a+b',
		testcases: [
			{ inputData: '1 2\n', expectedOutput: '3\n', isHidden: false },
			{ inputData: '-5 10\n', expectedOutput: '5\n', isHidden: true },
		],
	},
	{
		title: 'Factorial',
		description: 'Compute n! for 0 <= n <= 12.',
		inputFormat: 'Integer n',
		outputFormat: 'n!',
		testcases: [
			{ inputData: '5\n', expectedOutput: '120\n', isHidden: false },
			{ inputData: '0\n', expectedOutput: '1\n', isHidden: true },
		],
	},
	{
		title: 'Palindrome String',
		description: 'Check if a string is a palindrome (case-sensitive).',
		inputFormat: 'Single string s',
		outputFormat: 'YES or NO',
		testcases: [
			{ inputData: 'aba\n', expectedOutput: 'YES\n', isHidden: false },
			{ inputData: 'abc\n', expectedOutput: 'NO\n', isHidden: true },
		],
	},
	{
		title: 'Maximum of Array',
		description: 'Given n and an array of n integers, output the maximum.',
		inputFormat: 'n then n integers',
		outputFormat: 'max value',
		testcases: [
			{ inputData: '5\n1 3 2 9 4\n', expectedOutput: '9\n', isHidden: false },
			{ inputData: '3\n-5 -2 -9\n', expectedOutput: '-2\n', isHidden: true },
		],
	},
	{
		title: 'Fibonacci',
		description: 'Output the n-th Fibonacci number (0-indexed) for 0<=n<=30.',
		inputFormat: 'Integer n',
		outputFormat: 'F(n)',
		testcases: [
			{ inputData: '7\n', expectedOutput: '13\n', isHidden: false },
			{ inputData: '10\n', expectedOutput: '55\n', isHidden: true },
		],
	},
]

async function run() {
	const MONGO_URL = process.env.MONGO_URL || 'mongodb://mongo:27017/oj'
	await mongoose.connect(MONGO_URL)
	await Problem.deleteMany({})
	await Problem.insertMany(problems)
	console.log('Seeded 5 problems')
	await mongoose.disconnect()
}

run().catch(err => { console.error(err); process.exit(1) })

File: scripts/seed.py
import os
from app.database import SessionLocal, engine, Base
from app.models import Problem, TestCase

problems = [
	{
		"title": "A + B",
		"description": "Read two integers and output their sum.",
		"input_format": "Two integers a and b",
		"output_format": "Single integer a+b",
		"tests": [
			{"in": "1 2\n", "out": "3\n", "hidden": False},
			{"in": "-5 10\n", "out": "5\n", "hidden": True},
		],
	},
	{
		"title": "Factorial",
		"description": "Compute n! for 0 <= n <= 12.",
		"input_format": "Integer n",
		"output_format": "n!",
		"tests": [
			{"in": "5\n", "out": "120\n", "hidden": False},
			{"in": "0\n", "out": "1\n", "hidden": True},
		],
	},
	{
		"title": "Palindrome String",
		"description": "Check if a string is a palindrome (case-sensitive).",
		"input_format": "Single string s",
		"output_format": "YES or NO",
		"tests": [
			{"in": "aba\n", "out": "YES\n", "hidden": False},
			{"in": "abc\n", "out": "NO\n", "hidden": True},
		],
	},
	{
		"title": "Maximum of Array",
		"description": "Given n and an array of n integers, output the maximum.",
		"input_format": "n then n integers",
		"output_format": "max value",
		"tests": [
			{"in": "5\n1 3 2 9 4\n", "out": "9\n", "hidden": False},
			{"in": "3\n-5 -2 -9\n", "out": "-2\n", "hidden": True},
		],
	},
	{
		"title": "Fibonacci",
		"description": "Output the n-th Fibonacci number (0-indexed) for 0<=n<=30.",
		"input_format": "Integer n",
		"output_format": "F(n)",
		"tests": [
			{"in": "7\n", "out": "13\n", "hidden": False},
			{"in": "10\n", "out": "55\n", "hidden": True},
		],
	},
]


def run():
	Base.metadata.create_all(bind=engine)
	b = SessionLocal()
	try:
		for p in problems:
			prob = Problem(
				title=p["title"],
				description=p["description"],
				input_format=p["input_format"],
				output_format=p["output_format"],
			)
			b.add(prob)
			b.flush()
			for t in p["tests"]:
				b.add(TestCase(problem_id=prob.id, input_data=t["in"], expected_output=t["out"], is_hidden=1 if t["hidden"] else 0))
			b.commit()
	finally:
		b.close()


if __name__ == "__main__":
	run()







